engine/src/main/java/org/hibernate/search/store/impl/FSDirectoryProvider.java

```diff
@@ -79,24 +78,12 @@ public FSDirectory getDirectory() {
 
 	@Override
 	public boolean equals(Object obj) {
-		// this code is actually broken since the value change after initialize call
-		// but from a practical POV this is fine since we only call this method
-		// after initialize call
-		if ( obj == this ) {
-			return true;
-		}
-		if ( obj == null || !( obj instanceof FSDirectoryProvider ) ) {
-			return false;
-		}
-		return indexName.equals( ( (FSDirectoryProvider) obj ).indexName );
+		throw new AssertionFailure( "this type can not be compared reliably" );
```

Comment:
We used to need this, so for that reason they had "reasonable" implementations.

That requirement is gone - since many years a I suspect - and since it's damn complex to do it "right" I prefer to introduce the assertion to make sure we never rely on it.